native/src/main/java/io/ballerina/stdlib/file/service/FSListener.java

```diff
@@ -41,14 +42,16 @@ public FSListener(Runtime runtime) {
 
     @Override
     public void onMessage(LocalFileSystemEvent fileEvent) {
-        for (Map.Entry<BObject, Map<String, MethodType>> serviceEntry: serviceRegistry.entrySet()) {
-            MethodType serviceFunction = serviceEntry.getValue().get(fileEvent.getEvent());
-            if (serviceFunction != null) {
-                String functionName = serviceFunction.getName();
-                BObject service  = serviceEntry.getKey();
-                runtime.callMethod(service, functionName, null);
+        Thread.startVirtualThread(() -> {
+            for (Map.Entry<BObject, Map<String, MethodType>> serviceEntry: serviceRegistry.entrySet()) {
+                MethodType serviceFunction = serviceEntry.getValue().get(fileEvent.getEvent());
+                if (serviceFunction != null) {
+                    String functionName = serviceFunction.getName();
+                    BObject service  = serviceEntry.getKey();
+                    runtime.callMethod(service, functionName, new StrandMetadata(true, null));
```

Comment:
we should provide strandmetadata as below
    boolean isIsolated = serviceType.isIsolated() && serviceType.isIsolated(resourceName);
            StrandMetadata metaData = new StrandMetadata(isIsolated, properties);

Reply:
Fixed in f5374d02115ba2ebc0b0b848470ffa38f83463ab